src/core/utils/hooks/useTheme.ts

```diff
@@ -12,42 +12,47 @@ export type ThemeType = 'light' | 'dark' | 'os';
  * Defaults to light theme if none provided or set elsewhere.
  * @param theme Light, dark, or based on OS setting.
  */
-export const useTheme = (theme?: ThemeType): void => {
+export const useTheme = (
+  theme?: ThemeType,
+  owningDocument = document,
+): void => {
   React.useLayoutEffect(() => {
-    if (!document.body.classList.contains('iui-body')) {
-      document.body.classList.add('iui-body');
+    if (!owningDocument.body.classList.contains('iui-body')) {
+      owningDocument.body.classList.add('iui-body');
     }
-  }, []);
+  }, [owningDocument]);
 
   React.useLayoutEffect(() => {
     switch (theme) {
       case 'light':
-        addLightTheme();
+        addLightTheme(owningDocument);
         break;
       case 'dark':
-        addDarkTheme();
+        addDarkTheme(owningDocument);
         break;
       case 'os':
         if (window.matchMedia?.('(prefers-color-scheme: dark)').matches) {
-          addDarkTheme();
+          addDarkTheme(owningDocument);
         } else {
-          addLightTheme();
+          addLightTheme(owningDocument);
         }
         break;
       default:
-        if (!document.documentElement.classList.value.includes('iui-theme')) {
-          addLightTheme();
+        if (
+          !owningDocument.documentElement.classList.value.includes('iui-theme')
```

Comment:
@veekeys Should we change `value.includes` to `contains` here to avoid conflicts with your PR?

Reply:
Its fine, Ill merge if needed.